pkg/rpc/context.go

```diff
@@ -406,6 +405,11 @@ func (c ContextOptions) validate() error {
 	if c.Settings == nil {
 		return errors.New("Settings must be set")
 	}
+
+	// NB: OnSendPing and OnHandlePing default to noops.
+	// This is used both for testing and the cli.
+	_, _ = c.OnSendPing, c.OnHandlePing
```

Comment:
Could we just set them to noops here, during validation/setting defaults, instead of down below?